src/rust/aggregator/src/data/user_history.rs

```diff
@@ -70,7 +70,7 @@ impl Data for UserHistory {
         .fetch_all(&mut transaction as &mut PgConnection)
         .await
         .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
-        let mut change_events = sqlx::query!(
+        let change_events = sqlx::query!(
```

Comment:
Change size increase sends the order to the back of its price time priority queue. Implement the book state as a view the way I suggested and you can account for this by updating the order's timestamp.